src/runtime.ts

```diff
@@ -631,6 +586,143 @@ export async function createDefaultRuntime(
+        if (runtimeConfig.useIdentity) {
+          if (!contextLessRuntime) {
+            // create a runtime without account relation, just for accesing some bcc class instances
+            contextLessRuntime = createRuntime(
+              web3,
+              dfs,
+              {
+                accountMap: {
+                  [nullAddress]: 'no-private-key-here',
+                },
+              },
+              options,
+            );
+          }
+
+          // try out several dataKey encryption salting methods and check if, generated dataKeys are
+          // correct
+          let identity = nullAddress;
+          try {
+            identity = await (await contextLessRuntime).verifications
+              .getIdentityForAccount(accountId, true);
+          } catch (ex) {
+            // when no identity for a account could be found, it's a uninitialized account, so we
+            // should use default data key
+            useDefaultDatakey = true;
+            logger.log(`Could not find identity for account ${accountId} during keyConfig resolve.`
+              + ' Using default data key.', 'warning');
+          }
+
+          if (identity !== nullAddress) {
+            const encryptionSalts = [identity, accountId];
+            // eslint-disable-next-line guard-for-in
+            for (const encryptionSalt of encryptionSalts) {
+              const saltedDataKey = web3.utils
+                .keccak256(encryptionSalt + runtimeConfig.keyConfig[accountId])
+                .replace(/0x/g, '');
+              const tempRuntime = await createRuntime(web3, dfs, {
+                accountMap: {
+                  [accountId]: runtimeConfig.accountMap[accountId],
+                },
+                keyConfig: {
+                  [sha3Account]: saltedDataKey,
+                  [sha9Account]: saltedDataKey,
+                },
+                useIdentity: runtimeConfig.useIdentity,
+              });
+
+              if (tempRuntime.profile) {
+                dataKey = saltedDataKey;
+                logger.log(`Using encryptionSalt "${encryptionSalt}" for`
+                  + ` generating dataKey in keyConfig for account "${accountId}".`, 'debug');
+                break;
+              }
+            }
+          }
+        }
+
+        if (useDefaultDatakey) {
+          dataKey = web3.utils
+            .keccak256(accountId + runtimeConfig.keyConfig[accountId])
+            .replace(/0x/g, '');
+          logger.log('Using default encryptionSalt for generating dataKey'
+            + ` in keyConfig for account "${accountId}".`, 'debug');
+        } else if (!dataKey) {
+          throw new Error(`incorrect password for ${accountId} passed to keyConfig`);
+        }
+
+        // now add the different hashed accountids and datakeys to the runtimeconfig, if the
+        // dataKey was correct
+        runtimeConfig.keyConfig[sha3Account] = dataKey;
```

Comment:
Why do we need two configs with the same payload?

Reply:
It's because of the keyProvider internal logic. Normally, data keys for contracts are store for "edges". So this data key is reliable for me, so from me, to me: `sha3(account) => sha3(account)`.

If their is a contract data key, it would look like: `sha3(account) => sha3(contract)`.